### PR TITLE
[WIP] Excludes kube-system namespace webhook

### PR DESCRIPTION
**Pull request by rewanthtammana**
### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Test on WC
- [ ] Test on MC

### CODE REVIEW

File: helm/kyverno/values.yaml [hunk 1]
```diff
@@ -63,6 +63,21 @@ kyverno:
...
+          values:
+            - random
+            - asdf
+  # admissionWebhook:
+  #   config:
+  #     kind: ValidatingWebhookConfiguration
+  #     apiVersion: admissionregistration.k8s.io/v1beta1
+  #     excludeNamespaces:
+  #     - random
+  #     - qwer
```
Comment (pipo02mix) [inline helm/kyverno/values.yaml:80]:
```suggestion
  config:
    webhooks:
      - namespaceSelector:
          matchExpressions:
          - key: kubernetes.io/metadata.name
            operator: NotIn
            values:
              - random
              - asdf
    # admissionWebhook:
    #   config:
    #     kind: ValidatingWebhookConfiguration
    #     apiVersion: admissionregistration.k8s.io/v1beta1
    #     excludeNamespaces:
    #     - random
    #     - qwer
```

File: helm/kyverno/values.yaml [hunk 2]
```diff
@@ -63,6 +63,14 @@ kyverno:
...
     minAvailable: ""
   priorityClassName: "giantswarm-critical"
+  config:
+    webhooks:
+      - namespaceSelector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: NotIn
+            values:
+              - kube-system
```
[the remark this answers is not in the document]
Comment (rewanthtammana) [reply helm/kyverno/values.yaml:73]: Yes or should we enable [excludeKyvernoNamespace](https://github.com/giantswarm/kyverno-app/blob/main/helm/kyverno/charts/kyverno/values.yaml#L283-L285) flag? It excludes kyverno installed namespace in webhooks and resourceFilters. If `kyverno` gets installed in `random` namespace, then `random` namespace will be excluded automatically.

[the remark this answers is not in the document]
Comment (stone-z) [reply helm/kyverno/values.yaml:73]: That _seems_ like it should also work and is conveniently set by default. Just double check in your tests that both namespaces are present in the actual webhook configuration :pray:

[the remark this answers is not in the document]
Comment (rewanthtammana) [reply helm/kyverno/values.yaml:73]: Sure

[the remark this answers is not in the document]
Comment (rewanthtammana) [reply helm/kyverno/values.yaml:73]: The `excludeKyvernoNamespace` flag enables exclusion for the namespace in `resourceFilters` as well. Is that okay?

[the remark this answers is not in the document]
Comment (stone-z) [reply helm/kyverno/values.yaml:73]: Yeah that's fine. A technicality really if excluding at the webhook level

**Comment by rewanthtammana** [comment]
Fixes https://github.com/giantswarm/giantswarm/issues/24707

**Review by stone-z — APPROVED** [review]

**Pull request merged**